src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -263,17 +281,18 @@ public static CompactTree Create(LowLevelTransaction llt, Tree parent, Slice nam
             {
                 if (llt.Flags != TransactionFlags.ReadWrite)
                     return null;
-                
-                // TODO: This can be created a single time and stored in the root tree. 
-                var dictionaryId = PersistentHopeDictionary.CreateEmpty(llt);
+
+                // This will be created a single time and stored in the root page. 
+                var dictionaryId = PersistentDictionary.CreateDefault(llt);
```

Comment:
Is that per Voron instance, or CompactTree instance?

Reply:
The default is for Voron, but we store one per compact tree anyways as it is the chain head. I am not removing dictionaries BUT we are leaving the possibility for the following:
- Remove unused dictionaries if we ever need to. (That's why it is chain of dictionaries)
- That we could have 'terms' and 'index keys' tailored default dictionaries bootstrapped differently.

Reply:
What do you mean, storing one per key?
The issue here is that a CompactTree is a per term value, and if we generate the dictionaries on a per term basis, we'll have a LOT of them, no?

Reply:
I don't think I am understanding the question. Let me explain. Certain terms will be word types, those are going to be compressed very much from the get go. Now, we could have lots of different ways to partition terms, and get terms that are say GUIDs which have an entirely different structure. Those are not going to be stored so efficiently with the default dictionary. We could have tailored dictionaries if we would know in advance what goes there, like for instance in the map-reduce trees, we could use the knowledge that there are going to be storing GUIDs as keys, and act accordingly. 

The idea is to avoid having too many 'active' dictionaries, that is why the upgrading process could take care of that. That we are not deleting them (doing the housekeeping) is just something we dont want to introduce because of complexity vs cost tradeoff; but we aim to have as small amount of dictionaries as humanly possible. So dictionaries will be created on a per-tree basis; and pages upgraded when performing usual operations. Merging and splitting where we have to create a new page anyways, is a no brainer. However, on other operations like adding keys or even defrag, is where we should look into what makes sense in runtime. 

For example, the root page may not see much action over the lifetime of the tree, so most likely wont be upgraded UNLESS we need to execute a split operation. Would it make sense to upgrade on defrag? Probably.

Reply:
The key issue here is that you are looking at things at the _tree_ level, which is the equivalent of a single field, no?

If I have an index with 20 fields, I'll have 20 `CompactTree` fields, no?
And each one of them would have a separate _copy_ of the default dictionary (same for all) and then specialize those down the line?

What happens when we have an index with dynamic fields?

Reply:
That would depend on how we want to implement dynamic fields. Somehow thinking of having a tree per dynamic field sounds like a pretty wasteful thing. I would move toward a ```{prefix}\{term}``` encoding for those.

Reply:
Nope, we need it to be stored. We could have 'stored' a single instance in the 'root' tree. [Sounds reasonable to do it like that] But we need it to be stored because potentially every build of RavenDB could come with a different dictionary (not that it would, but it could happen). So better to have a hard copy of it to avoid having to deal with that.

Reply:
What is?

Reply:
I am implementing the ability to share the default dictionary.